src/main/java/net/neoforged/neoforge/fluids/potion/PotionFluidType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids.potion;
+
+import java.util.function.Consumer;
+import net.minecraft.core.component.DataComponents;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.alchemy.Potion;
+import net.minecraft.world.item.alchemy.PotionContents;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.fluids.FluidStack;
+import net.neoforged.neoforge.fluids.FluidType;
+
+public class PotionFluidType extends FluidType {
+    public PotionFluidType(Properties properties) {
+        super(properties);
+    }
+
+    @Override
+    public String getDescriptionId(FluidStack stack) {
+        var potion = stack.getOrDefault(DataComponents.POTION_CONTENTS, PotionContents.EMPTY).potion();
+        return Potion.getName(potion, "%s.effect.".formatted(Items.POTION.getDescriptionId()));
+    }
+
+    @Override
+    public ItemStack getBucket(FluidStack stack) {
+        // if theres at least 1 bucket, we use the new potion bucket item
+        // otherwise return a potion bottle
```

Comment:
That is not what this code is doing. This code says if there is an exact multiple of buckets use bucket, otherwise use potion.